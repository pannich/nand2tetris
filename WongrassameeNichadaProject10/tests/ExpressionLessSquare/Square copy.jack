// This file is part of www.nand2tetris.org
// and the book "The Elements of Computing Systems"
// by Nisan and Schocken, MIT Press.
/// File name: projects/10/ExpressionLessSquare/Square.jack

/** Expressionless version of projects/10/Square/Square.jack. */

class Square {

   field int x, y;
   field int size;

   constructor Square new(int Ax, int Ay, int Asize) {
      let x = Ax;
      let y = Ay;
      let size = Asize;
      do draw();
      return x;
   }

   method void incSize() {
      if (x) {
         do erase();
         let size = size;
         do draw();
      }
      return;
   }
}
