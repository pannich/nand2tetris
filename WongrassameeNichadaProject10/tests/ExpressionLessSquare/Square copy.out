class Square {
field int x, y;
field int size;
constructor Square new(int Ax, int Ay, int Asize) {
let x = Ax;
let y = Ay;
let size = Asize;
do draw();
return x;
}
method void incSize() {
if (x) {
do erase();
let size = size;
do draw();
}
return;
}
}
